core/src/network/direct_message.rs

```diff
@@ -1,4 +1,18 @@
-use holochain_core_types::{cas::content::Address, validation::ValidationPackage};
+use holochain_core_types::{
+    cas::content::Address, error::HolochainError, json::JsonString, validation::ValidationPackage,
+};
+
+/// This is direct message that got created by the zome code through hdk::send().
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug, DefaultJson)]
+pub struct CustomDirectMessage {
+    /// We have to track which zome sent the message so we can call the
```

Comment:
That's an interesting point... so an app can have one `receive` callback per zome? 

But it could handle different event types, if they were passed as data, and using a `match` in receive?

Reply:
@Connoropolous pretty similar to how websockets works i guess, you can send and receive strings, but dispatching logic and simulating request/response behaviour is BYO